R/net.mod.trackers.R

```diff
@@ -0,0 +1,119 @@
+#' @title Trackers: netsim Module
+#'
+#' @description This function apply the user provided epi trackers
+#'
+#' @param dat Master list object containing a \code{networkDynamic} object and
+#'        other initialization information passed from \code{\link{netsim}}.
+#' @param at Current time step.
+#'
+#' @return the updated dat Master list object.
+#'
+#' @section Optional Module:
+#' This module is not included by default
+#'
+#' @section The \code{tracker.list} list:
+#' \code{tracker.list} is a list of NAMED functions stored in the \code{param}
+#' list of the \code{dat} master list object.
+#'
+#' @section Tracker Functions:
+#' This module will apply the tracker functions present in the parameter list
+#' \code{tracker.list}. Each tracker must be a function with EXACTLY two
+#' arguments: the \code{dat} Master list object and \code{at} the current time
+#' step. They must return a VALUE of length one (numeric, logical or character).
+#'
+#' @examples
+#' \dontrun{
+#'
+#' # Create some trackers
+#' epi_prop_infected <- function(dat, at) {
+#'   needed_attributes <- c("status", "active")
+#'   output <- with(get_attr_list(dat, needed_attributes), {
+#'     pop <- active == 1
+#'     cond <- status == "i"
+#'
+#'     out <- sum(cond & pop, na.rm = TRUE) / sum(pop, na.rm = TRUE)
+#'
+#'     out
+#'   })
+#'
+#'   return(output)
+#' }
+#'
+#' epi_s_num <- function(dat, at) {
+#'   needed_attributes <- c("status")
+#'   output <- with(get_attr_list(dat, needed_attributes), {
+#'     out <- sum(status == "s", na.rm = TRUE)
+#'
+#'     out
+#'   })
+#'
+#'   return(output)
+#' }
+#'
+#' # Create the `tracker.list` list
+#' tracker.list <- list(
+#'   prop_infected = epi_prop_infected,
+#'   s_num         = epi_s_num
+#' )
+#'
+#'  # Do not forget to add it to `param`
+#'  param <- param.net(
+#'    inf.prob = 0.3,
+#'    act.rate = 0.5,
+#'    tracker.list = tracker.list
```

Comment:
I feel like this custom tracker is more of a "control" rather than a "parameter", since in effect it is modifying the prevalence module defined in the controls. Feel free to convince me otherwise. :)

Reply:
i.e., it should go under `control.net` rather than `param.net`

Reply:
That was also my initial idea but I was unsure of exactly what defines a "parameter" and a "control". I don't even remember what was my reasoning for choosing `param.net` and not `control.net`. I'll switch